modules/ballerina-lang/src/main/java/org/ballerinalang/model/tree/FunctionNode.java

```diff
@@ -21,4 +21,13 @@
  * @since 0.94
  */
 public interface FunctionNode extends InvokableNode, AnnotatableNode, TopLevelNode {
+
+  boolean isDeclaredWithType();
```

Comment:
This is usually called receiver in programming languages.  Following two methods are enough.

VariableNode getReceiver ();

void setReceiver (VariableNode receiverNode);